### PR TITLE
Assignment 1 + Lab (1,2 already updated) and 3

### PR DESCRIPTION
**Pull request by nthrsn**
Note: Extension granted by learning support (Tianyi) to submit by Friday - thank you!

## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
A1 notebook and lab code (1-3)

## What did you learn from the changes you have made?
Training models, how to use optimizers

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
N/A - followed lectures best as I can

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
Concepts took a while to understand

## How were these changes tested?
Colab

## Checklist
- [x] I can confirm that my changes are working as intended

### CODE REVIEW

File: 02_activities/assignments/assignment_1.ipynb
Comment (tianyi21) [inline 02_activities/assignments/assignment_1.ipynb:1]: - Task 2a: In each conv block, you added two same `Conv2D`. If you add only one instead for each block, the performance will become better.
- Task 3a: Instead of using `x_test`, `y_test` for training, please use the `x_val`, `y_val` you created in 1b and reserve test data for testing purpose.
- Task 3a: Your validation accuracy stays at `0.01`, ie 1% for 100 class classification, which means your model isn't learning. I took a closer look and it turns out there were some issues with your model implementation in Task 2a.

**Review by tianyi21 — APPROVED** [review]
Hi Nithursan!

Thank you for submitting your DL assignments. Please refer to my comments attached to each file.

Your A1 and Lab 3 are now complete.

Thanks,
Tianyi [LS]

File: 01_materials/labs/lab_1.ipynb
Comment (tianyi21) [inline 01_materials/labs/lab_1.ipynb:1]: - b: Please always reinitialize your model with `model = Sequential() ...` to reset all model parameters in addition to `model.compile()`. Or, you are continuing the training from the previous trained state with a different model setup. You may check this with the accuracy of the first epoch.

**Review by tianyi21 — APPROVED** [review]
Hi Nithursan!

Thank you for submitting your DL assignments. Please refer to my comments attached to each file.

Your Lab 1, 2 are now complete.

Thanks,
Tianyi [LS]
